### ISSUE TITLE
Elixir 1.6.3: Compile error in Hub.subscribe expression

### ISSUE THREAD

**Issue by TrailingDots** [wise-home/hub#34]
Attempting to follow the README with the example and I am getting
a compile error:

cecilm@metali2 2 hub_demo$ mix test
Compiling 1 file (.ex)

== Compilation error in file lib/hub_demo/demo.ex ==
** (CompileError) lib/hub_demo/demo.ex:10: invalid use of _. "_" represents a value to be ignored in a pattern and cannot be used in expressions
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
    (stdlib) lists.erl:1355: :lists.mapfoldl/3
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
cecilm@metali2 2 hub_demo$ mix --version
Erlang/OTP 20 [erts-9.2] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:10] [hipe] [kernel-poll:false]

Mix 1.6.3 (compiled with OTP 20)

The problem line is
```
    Hub.subscribe("some_channel1", %{name: _, age: age} when age > 42)
```

I used the latest version in github.com.

Here's the dumb test:

```
defmodule HubDemo.Demo do
  @moduledoc """
  Demo as per the github on
  https://github.com/wise-home/hub
  """
  require Logger

  def greet() do
    Logger.info("starting greet")
    Hub.subscribe("some_channel1", %{name: _, age: age} when age > 42)
    loop()
    end
  
  def loop()
    receive do
      %{name: name} -> Logger.info("#{name} is > 42")
    end
    loop()
  end
end

defmodule HubDemo.StartDemo do
  def demo_start() do
    Logger.info("starting demo_start")
    greet = spawn(HubDemo.Demo, :greet, [])
    Hub.publish("some_channel1", %{name: "John", age: 48})
    Hub.publish("some_channel1", %{name: "Alex", age: 9})
    Hub.publish("some_channel1", %{name: "Anna", age: 43})
  end
end
```

**Comment by TrailingDots** [comment]
Same error with your slightly moded script. I knew the script contained
minor problems,
but if the compiler breaks, ...  :)  :(

I have erlang 20.2. Will upgrade to 20.3.8.6 or later and try it.

Thanks for the rapid reply,

Cecil McGregor

**Comment by TrailingDots** [comment]
I just compiled erlang 20.3 and it works!  The download page doesn't detail
beyond 20.3 and I couldn't find
a more detailed version. I suspect the official version just bump minor
builds.

I tried to compile erlang 21.0 but got some errors. Will look into this
later.

BTW, using Elixir 1.6.3 for all this code.

This was a strange compiler error. I don't know the root cause, but thanks
for looking at it.
